src/rust/engine/src/externs.rs

```diff
@@ -87,6 +87,10 @@ impl Externs {
     (self.val_for)(self.context, key)
   }
 
+  pub fn val_for_id(&self, id: Id) -> Value {
+    self.val_for(&Key::new_with_anon_type_id(id))
```

Comment:
Hm. By that do you mean it'd be worth doing as part of this review, or can it wait as another small change?